pkg/awsutils/awsutils.go

```diff
@@ -703,13 +706,16 @@ func (cache *EC2InstanceMetadataCache) awsGetFreeDeviceNumber() (int, error) {
 	inst := result.Reservations[0].Instances[0]
 	var device [maxENIs]bool
 	for _, eni := range inst.NetworkInterfaces {
-		if aws.Int64Value(eni.Attachment.DeviceIndex) > maxENIs {
-			log.Warnf("The Device Index %d of the attached ENI %s > instance max slot %d",
-				aws.Int64Value(eni.Attachment.DeviceIndex), aws.StringValue(eni.NetworkInterfaceId),
-				maxENIs)
-		} else {
-			log.Debugf("Discovered device number is used: %d", aws.Int64Value(eni.Attachment.DeviceIndex))
-			device[aws.Int64Value(eni.Attachment.DeviceIndex)] = true
+		// We don't support multi-card yet, so only account for network card zero
+		if aws.Int64Value(eni.Attachment.NetworkCardIndex) == 0 {
```

Comment:
Do we still need this change?

Reply:
Yep so we don't mark a device index on card 0 as used when it is really not.

Reply:
Ahh right since we do a describe call..later on we can actually enhance it..should be fine for now..